cylc/flow/scripts/lint.py

```diff
@@ -54,8 +55,142 @@
 from cylc.flow.cfgspec.workflow import upg, SPEC
 from cylc.flow.id_cli import parse_id
 from cylc.flow.parsec.config import ParsecConfig
+from cylc.flow.scripts.cylc import DEAD_ENDS
 from cylc.flow.terminal import cli_function
 
+DEPRECATED_ENV_VARS = {
+    'CYLC_SUITE_HOST': 'CYLC_WORKFLOW_HOST',
+    'CYLC_SUITE_OWNER': 'CYLC_WORKFLOW_OWNER',
+    'CYLC_SUITE_SHARE_DIR': 'CYLC_WORKFLOW_SHARE_DIR',
+    'CYLC_SUITE_SHARE_PATH': 'CYLC_WORKFLOW_SHARE_PATH',
+    'CYLC_SUITE_NAME': 'CYLC_WORKFLOW_ID',
+    'CYLC_SUITE_LOG_DIR': 'CYLC_WORKFLOW_LOG_DIR',
+    'CYLC_SUITE_INITIAL_CYCLE_POINT': 'CYLC_WORKFLOW_INITIAL_CYCLE_POINT',
+    'CYLC_SUITE_INITIAL_CYCLE_TIME': 'CYLC_WORKFLOW_INITIAL_CYCLE_TIME',
+    'CYLC_SUITE_FINAL_CYCLE_POINT': 'CYLC_WORKFLOW_FINAL_CYCLE_POINT',
+    'CYLC_SUITE_FINAL_CYCLE_TIME': 'CYLC_WORKFLOW_FINAL_CYCLE_TIME',
+    'CYLC_SUITE_WORK_DIR': 'CYLC_WORKFLOW_WORK_DIR',
+    'CYLC_SUITE_UUID': 'CYLC_WORKFLOW_UUID',
+    'CYLC_SUITE_RUN_DIR': 'CYLC_WORKFLOW_RUN_DIR',
+}
+
+OBSOLETE_ENV_VARS = {
+    'CYLC_SUITE_DEF_PATH',
+    'CYLC_SUITE_DEF_PATH_ON_SUITE_HOST'
+}
+
+
+def check_jinja2_no_shebang(line, **kwargs):
```

Comment:
Why use a `kwargs` dict instead of explicitly specifying the args?
```suggestion
def check_jinja2_no_shebang(line, file, jinja_shebang=False, fallback=None):
```

Reply:
Because these functions are called generically with a larger set of kwargs - some of which this one doesn't need.

Reply:
```suggestion
def check_jinja2_no_shebang(line, file, jinja_shebang=False, fallback=None, **kwargs):
```